dot/node.go

```diff
@@ -354,6 +391,34 @@ func setupMetricsServer(address string) {
 	}()
 }
 
+// stores the global node name to reuse
+func storeGlobalNodeName(name, basepath string) error {
+	db, err := state.SetupDatabase(basepath)
+	if err != nil {
+		return err
+	}
+
+	basestate := state.NewBaseState(db)
+	err = basestate.StoreNodeGlobalName(name)
+	if err != nil {
+		logger.Warn(
+			"failed to store global node name",
+			"basepath", basepath,
+			"error", err,
+		)
+		return nil
+	}
+
+	// close database
+	err = db.Close()
```

Comment:
db should be closed whether or not the node global name was stored.  I would use a `defer`.